src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -716,12 +716,12 @@ export function getBracketsToClose(text: string) {
       }
 
       const substr = text.slice(i, i + openBracket.length);
-      if (substr === openBracket) {
-        stack.push(substr);
-        break;
-      } else if (pairsReversed[substr] && pairsReversed[substr] === stack[stack.length - 1]) {
+      if (pairsReversed[substr] && pairsReversed[substr] === stack[stack.length - 1]) {
         stack.pop();
         break;
+      } else if (substr === openBracket) {
+        stack.push(substr);
+        break;
```

Comment:
This needed to be reversed. `"""` is a symmetric token (both opening and closing) so the algo needs to check whether one has already been seen at the corresponding level to know how to interpret it.